packages/core/admin/admin/src/content-manager/components/BlocksInput/BlocksToolbar.tsx

```diff
@@ -418,14 +353,48 @@ const ListButton = ({ block, format }: ListButtonProps) => {
     return false;
   };
 
+  const toggleList = (format: Block<'list'>['format']) => {
+    let currentListEntry;
+    if (editor.selection)
+      currentListEntry = Editor.above(editor, {
+        match: (node) => !Editor.isEditor(node) && node.type === 'list',
+      });
+    else {
+      // If no selection, toggle last inserted node
+      const [_, lastNodePath] = Editor.last(editor, []);
+      currentListEntry = Editor.above(editor, {
+        match: (node) => !Editor.isEditor(node) && node.type === 'list',
+        at: lastNodePath,
+      });
+    }
+
+    // If selection is already a list then toggle format
+    if (currentListEntry) {
+      const [currentList, currentListPath] = currentListEntry;
+
+      if (!Editor.isEditor(currentList) && isListNode(currentList)) {
+        if (currentList.format !== format) {
+          // Format is different, toggle list format
+          Transforms.setNodes(editor, { format }, { at: currentListPath });
+        } else {
+          // Format is same, convert selected list-item to paragraph
+          blocks['paragraph'].handleConvert!(editor);
+        }
+      }
+    } else {
+      // If selection is not a list then convert it to list
+      blocks[`list-${format}`].handleConvert!(editor);
+    }
```

Comment:
I think the code would be more straight forward if you do an early return for this case instead of the if/else